### PR TITLE
Add a basic debug interface for the multichain

### PR DESCRIPTION
**Pull request by pimveldhuisen**
This adds basic multichain info in the debug panel.
I have not been super strict on the pep8, because:
 1) all the other wx stuff is using old style (e.g. CamelCase)
 2) The wx gui will be eliminated in the foreseeable future

### CODE REVIEW

**Comment by pimveldhuisen** [comment]
Retest this please

**Comment by pimveldhuisen** [comment]
Retest this please

**Comment by devos50** [comment]
Nice! Is there any way you can create a method in the multi chain to easily get a dictionary with statistics? We could use that one to create an endpoint to fetch multi chain statistics.

**Comment by pimveldhuisen** [comment]
@devos50 Yes, that seems like a good idea. So the best way to implement this would just be a regular method in community.py that returns a dict with stats?

**Comment by devos50** [comment]
@pimveldhuisen yes :)

**Comment by synctext** [comment]
looking forward to trying this out.

**Comment by pimveldhuisen** [comment]
@devos50 I'm working on your suggestion, and I think it works out well; it gives a much better seperation between gui and logic.
@synctext You can ;) This commit (4f5451f) should be functional as it is. It is part of the debug gui (ctrl/cmd-D)

**Comment by devos50** [comment]
Absolutely! I'm also refactoring several other statistic methods (I.e. In BarterStatistics) to return dictionaries instead of formatted strings. After that's, it's only a matter of a few lines to create a new endpoint to return this data and to display this data in the old and new GUI :).

Verstuurd vanaf mijn iPhone

**Comment by pimveldhuisen** [comment]
Retest this please

**Comment by pimveldhuisen** [comment]
Retest this please

**Comment by whirm** [comment]
@pimveldhuisen if something fails and it's not related to your PR, please comment on it so we can keep track

**Comment by pimveldhuisen** [comment]
On windows 64:
Tribler.Test.Core.Modules.RestApi.test_my_channel_endpoints.TestMyChannelModifyTorrentsEndpoint.test_add_torrent_from_url_to_my_channel_with_description (from nosetests)

Failing for the past 1 build (Since Failed#1465 )
Took 7.3 sec.
Error Message

Test raised 1 unhandled Twisted exceptions, last one was: Unhandled Error
Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Core\Utilities\twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "C:\Python27\lib\site-packages\twisted\internet\base.py", line 1194, in run
    self.mainLoop()
  File "C:\Python27\lib\site-packages\twisted\internet\base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "C:\Python27\lib\site-packages\twisted\internet\base.py", line 825, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Core\APIImplementation\threadpoolmanager.py", line 50, in caller
    fun(*args, **kwargs)
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Core\Libtorrent\LibtorrentMgr.py", line 482, in do_dht_check
    lt_session = self.get_session()
  File "C:\workspace\jenkins\workspace\GH_Tribler_PR_tests_win64\tribler\Tribler\Core\Libtorrent\LibtorrentMgr.py", line 186, in get_session
    if hops not in self.ltsessions:
exceptions.TypeError: argument of type 'NoneType' is not iterable

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
```test_anon_tunnel (Tribler.Test.test_tunnel_community_positive.TestTunnelCommunityPositive) ... nosetests: Fatal IO error 11 (Resource temporarily unavailable) on X server :18958.
```

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by whirm** [comment]
@pimveldhuisen maybe you want to cover `Tribler/community/multichain/community.py`? This is going to be used even after WX gets kicked out.

**Comment by whirm** [comment]
The rest looks fine

**Comment by pimveldhuisen** [comment]
```
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] OUT: processing 4_nosetests.xml.part
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] OUT: processing 5_nosetests.xml.part
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] OUT: processing 6_nosetests.xml.part
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: -:50.9999945: internal error: Huge input lookup
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR:    1466174056        endpoint:StandaloneEndpoint:209  1 came in, 177 bytes in to
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR:                                                                                ^
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: -:50.6: internal error: Huge input lookup
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: 
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: ^
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: Broken pipe: fwrite()
INFO:OneShotProcessProtocol:[run_nosetests_for_je...] ERR: flush error
```

**Comment by pimveldhuisen** [comment]
retest this please

**Comment by pimveldhuisen** [comment]
```
test_anon_download_exitnode_changeofmind (Tribler.Test.test_tunnel_community_negative.TestTunnelCommunityNegative) ... nosetests: Fatal IO error 11 (Resource temporarily unavailable) on X server :31648.
```

**Comment by pimveldhuisen** [comment]
retest this please

File: Tribler/community/multichain/database.py [hunk 1]
```diff
@@ -145,6 +145,9 @@ def get_by_hash(self, hash):
         :param hash: The hash of the block that needs to be retrieved.
         :return: The block that was requested or None
         """
+        if hash is None:
```
Comment (whirm) [inline Tribler/community/multichain/database.py]: Why is this checked here instead of avoiding calling this function at all if there's no hash?

Comment (pimveldhuisen) [inline Tribler/community/multichain/database.py]: @whirm I some scenario's the hash argument is in itself the result of a query, for example: 
https://github.com/Tribler/tribler/blob/devel/Tribler/community/multichain/database.py#L126
Of course the if None statement could be placed there, but I think this is both safer and more elegant.

**Pull request merged**